sklearn/manifold/t_sne.py

```diff
@@ -142,10 +142,12 @@ def _kl_divergence(params, P, degrees_of_freedom, n_samples, n_components,
 
     # Q is a heavy-tailed distribution: Student's t-distribution
     n = pdist(X_embedded, "sqeuclidean")
-    n += 1.
```

Comment:
By the way, yes you are correct about the wrong order of dividing by alpha and adding 1. Probably nobody used n_components > 2 before. :) We need some kind of unit test for that. In the best case we would compare the numerical and the analytical gradient.